cobaya/likelihoods/planck_2018_lowl/TT_native.py

```diff
@@ -0,0 +1,157 @@
+
+    @classmethod
+    def get_bibtex(cls):
+        from cobaya.likelihoods.planck_2018_lowl.TT import TT
+        return TT.get_bibtex()
+
+    def get_requirements(self):
+        return {'Cl': {'tt': self._lmax}}
+
+    def initialize(self):
+        if self.get_install_options() and self.packages_path:
+            path = self.get_path(self.packages_path)
+            self._data['cov'][2:, 2:] = np.loadtxt(os.path.join(path, 'cov.txt'))
+            self._data['cl2x'][:, 2:, 0] = np.loadtxt(os.path.join(path, 'cl2x_1.txt')) 
+            self._data['cl2x'][:, 2:, 1] = np.loadtxt(os.path.join(path, 'cl2x_2.txt')) 
+            self._data['cl2x'][:, 2:, 2] = np.loadtxt(os.path.join(path, 'cl2x_3.txt')) 
+            self._data['mu'][2:] = np.loadtxt(os.path.join(path, 'mu.txt'))
+            self._data['mu_sigma'][2:] = np.loadtxt(os.path.join(path, 'mu_sigma.txt'))
+            _nbins = len(self._data['cl2x'][:, 0, 0])
+
+            # Bandlimit covariance matrix
+            for l in range(self._lmin, self._lmax+1):
+                for k in range(self._lmin, self._lmax+1):
+                    if abs(l-k) > self._delta_l:
+                        self._data[cov][l, k] = 0
+
+            # Set up prior
+            self._data['prior'] = np.zeros((251, 2))
+            for l in range(self._lmin, self._lmax+1):
+                j = 0
+                while abs(self._data['cl2x'][j, l, 1] + 5) < 1e-4:
+                    j += 1
+                self._data['prior'][l, 0] = self._data['cl2x'][j+2, l, 0]
+                j = _nbins-1
+                while abs(self._data['cl2x'][j, l, 1] - 5) < 1e-4:
+                    j -= 1
+                self._data['prior'][l, 1] = self._data['cl2x'][j-2, l, 0]
+
+            np.savetxt('prior.txt', self._data['prior'])
+
+            self._data['offset'] = self.log_likelihood(self._data['mu_sigma'], init=True)
+                
+
+    def get_requirements(self):
+        return {'Cl': {'tt': self._lmax}}
+
+
+    def log_likelihood(self, cls_TT, init=False):
+
+        if (any(cls_TT[self._lmin:self._lmax+1] < self._data['prior'][self._lmin:self._lmax+1, 0]) or
+                any(cls_TT[self._lmin:self._lmax+1] > self._data['prior'][self._lmin:self._lmax+1, 1])):
+            return -1e30
+
+        # Convert the cl's to Gaussianized variables
+        x = np.zeros(self._lmax+1)
+        for l in range(self._lmin, self._lmax+1):
+            x[l] = self._splint_gauss_br(self._data['cl2x'][:, l, 0],
+                                        self._data['cl2x'][:, l, 1],
+                                        self._data['cl2x'][:, l, 2],
+                                        cls_TT[l])
+        logl = -0.5 * sum((x[self._lmin:self._lmax+1] - self._data['mu'][self._lmin:self._lmax+1]) * 
+                          np.dot(self._data['cov'][self._lmin:self._lmax+1, self._lmin:self._lmax+1],
+                                 (x[self._lmin:self._lmax+1] - self._data['mu'][self._lmin:self._lmax+1])))
+
```

Comment:
Calculate delta once and use numpy dot for both contractions? (covinv.dot(delta).dot(delta))